examples/2-tier-iis-sql-vm/example/main.tf

```diff
@@ -0,0 +1,102 @@
+# Configure the Microsoft Azure Provider
+provider "azurerm" {
+  subscription_id = "${var.subscription_id}"
+  client_id       = "${var.client_id}"
+  client_secret   = "${var.client_secret}"
+  tenant_id       = "${var.tenant_id}"
```

Comment:
Rather than specifying these variables in-line (which hard-codes this to a Service Principal) - if you leave these values unassigned Terraform will try and pull them either from the Environment Variables or from the Azure CLI - which means that these variables aren't as  likely to be accidentally checked-in to source control.

One thing that we'd strongly recommend doing in the Provider block however is pinning to a version, for example:

```
provider "azurerm" {
  version = "=1.17.0"
}
```